src/Microsoft.ML.Tokenizers/Utils/ByteArrayComparer.cs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.ML.Tokenizers
+{
+    internal class ByteArrayComparer : IEqualityComparer<byte[]>
+    {
+        public bool Equals(byte[] x, byte[] y)
+        {
+            if (x is null || y is null)
+            {
+                return x == y;
+            }
+
+            if (x.Length != y.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < x.Length; i++)
+            {
+                if (x[i] != y[i])
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        public int GetHashCode(byte[] bytes)
+        {
+            if (bytes == null)
+            {
+                throw new ArgumentNullException(nameof(bytes));
+            }
+
+            int hash = 17;
+            foreach (byte b in bytes)
+            {
+                hash = hash * 31 + b;
+            }
+
+            return hash;
```

Comment:
How long are the arrays this is typically hashing? If small, this is fine. If larger, we'd want to use something faster / vectorized.

Reply:
In common cases are short array. The array contains the UTF-8 encoded bytes of the tokens we read from the vocabulary files. token length are short.